src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -245,8 +253,22 @@ public async Task<ResourceEvent> WaitForResourceHealthyAsync(string resourceName
     /// <param name="waitBehavior">The wait behavior.</param>
     /// <returns>A task.</returns>
     /// <remarks>
+    /// <para>
     /// This method returns a task that will complete with the resource is healthy. A resource
-    /// without <see cref="HealthCheckAnnotation"/> annotations will be considered healthy.
+    /// without <see cref="HealthCheckAnnotation"/> annotations will be considered healthy. The
+    /// <see cref="WaitBehavior"/> controls how the wait operation behaves when the resource
+    /// enters an unavailable state such as <see cref="KnownResourceStates.FailedToStart"/>.
+    /// </para>
+    /// <para>
+    /// When <see cref="WaitBehavior.WaitOnResourceUnavailable"/> is specified the wait operation
+    /// then the method will continue to wait until the resource reaches a <see cref="KnownResourceStates.Running"/> state. This is the default
+    /// behavior with the <see cref="WaitForResourceHealthyAsync(string, CancellationToken)"/> overload.
```

Comment:
```suggestion
    /// then the method will continue to wait until the resource reaches a <see cref="KnownResourceStates.Running"/> state.
```

I would remove this sentence since:

1. It is about a different method than what we are documenting here.
2. It isn't accurate since the ResourceNotificationServiceOptions.DefaultWaitBehavior can (and will) change the behavior of that method.